jhu-idc/idc-isle-dc
# Metadata export

This provides the views for metadata export.  It integrates the `idc_export` module into the code. 
## General Usage
This PR introduces two new endpoints for exporting metadata through (in a CSV file format)

This one will return Repository items:

`https://islandora-idc.traefik.me/export_items`

This endpoint will return Collection Objects:

`https://islandora-idc.traefik.me/export_collections`

Both of these take search queries on the end of them like so: `query=zoo`  or `query=(its_field_member_of:33)`, where the second one is direct solr query.  These operates similarly to the search api View we already use.  

These trigger islandora batches to run, which will redirect on the screen a few times. There will be a link in the message box to click on to start a download of the resulting file.  It stops on the `/search` page - we might want to change this behavior if we can. 

## How to test this ticket

(all of these contain node_ids, which you'll have to adjust according to your system setup). 

  * Load the test data:   `make test test=20-export-tests`  (there are no tests yet, only migrations)
  * Run a few queries: 
    * `https://islandora-idc.traefik.me/export_items?query=zoo`  to get all repository items that have the word `zoo` in them. 
       *  compare results to what's in the original CSV file in the testcafe setup (or compare what's exported to what the search result has for the same search.) Ensure you got all the objects expected. 
    * `https://islandora-idc.traefik.me/export_collections?query=zoo` to get all collection objects that have the word `zoo` in them.  
      *  compare results to what's in the original CSV file in the testcafe setup (or compare what's exported to what the search result has for the same search). Ensure you got all the objects expected.  
    *  Get an export for one Repository Item: `https://islandora-traefik.me/export_items?query=(its_nid:123)` Ensure it's exported correctly. 
    * Get an export for all direct descendants of a collection:  `https://islandora-traefik.me/export_items?query=(its_field_member_of:123)`  Ensure that CSV of the repository items returned is complete and that all are direct descendants of the collection. 
    * Try a more complex query using facets - I'm a little unsure how it handles that scenario. 
    *  Run a round trip of the file: try changing data in one of the exported files (like title, or what not) and re-ingest it. Ensure the object(s) is updated in the system.  You do not need to do anything with the `node_id` or the `local_id` fields.`local_id` field should be empty. 
    

## Test Infrastructure setup 

This also provides the beginning code for testing export - there is a directory with migrations that are loaded via testcafe. 
What's missing are the export tests themselves, though it does load content to test with. 

## Notes
This PR also enables the Message box, because it's necessary to have that to get the file download link from metadata export.   Not sure we really want this and might need to change this part. 

Notes: 
  * all the `block.block*` config changes are due to enabling the message box. 

Addresses: 

https://github.com/jhu-idc/iDC-general/issues/326  
https://github.com/jhu-idc/iDC-general/issues/328
https://github.com/jhu-idc/idc-isle-dc/issues/119

Adds test infrastructure to start addressing (but doesn't finish): 
https://github.com/jhu-idc/iDC-general/issues/327
https://github.com/jhu-idc/iDC-general/issues/329
